docs/conf.py

```
@@ -80,3 +83,41 @@ def setup(app):
+    outdir = srcdir / GENERATED_DIRNAME
+    # avoid re-generating repeatedly
+    if outdir.exists():
+        shutil.rmtree(outdir)
+    outdir.mkdir(parents=True, exist_ok=True)
+
+    tmp = Path(tempfile.mkdtemp(prefix="nfdocs-"))
+    try:
+        # clone only the ref
+        subprocess.run(["git", "clone", "--depth", "1", "--branch", PIPELINE_REF, PIPELINE_GIT, str(tmp)], check=True)
+        # run nf-docs CLI command
+        subprocess.run(["nf-docs", "generate", str(tmp)], check=True)
```

Ruff (0.15.5)

[error] `subprocess` call: check for execution of untrusted input

(S603)